08_arrays/main.go

```diff
@@ -43,45 +43,213 @@ func main() {
 	// TODO: Fix the errors
 	var a []int
 	a = append(a, 0)
-	a = append(a, "1")
+	a = append(a, 1)
 	a = append(a, 2)
 	fmt.Println(a)
-	fmt.Println(a["0"])
+	fmt.Println(a[0])
 
 	// TODO: Create an array with 5 integers: 1, 2, 3, 4, 5
 	// Print the array.
+	var fiveInt []int
+	fiveInt = append(fiveInt, 1, 2, 3, 4, 5)
+	fmt.Println(fiveInt)
 
 	// TODO: Use a loop to insert 10 integers in an array.
 	// Print the array.
+	var tenInt []int
+	var count int = 1
+	var times int = 10
+
+	for {
+		if count > times {
+			break
+		}
+		tenInt = append(tenInt, count)
+		count++
+	}
+
+	fmt.Println(tenInt)
 
 	// TODO: Using the array from the previous step, print the first and the
 	// last elements
+	fmt.Println(tenInt[0])
+	fmt.Println(tenInt[len(tenInt)-1])
 
 	// TODO: Using the array from the previous step, create a new array
 	// containing the first 5 elements multiplied by 2.
 	// Hint: Use a loop.
 	// Hint: The new array should contain [2, 4, 6, 8, 10].
 	// Print the new array.
 
+	// create copy of array
+	// loop 5 times (first 4 elements needed)
+	// multiply elements * 2
+	// add elements to new array
+	// print new array
+	// var timesTwo []int = tenInt
+	var timesTwo [5]int
+	var count2 = 0
+
+	for {
+		if count2 > 4 {
+			break
+		}
+		timesTwo[count2] = (tenInt[count2] * 2)
+		count2++
+	}
+
+	fmt.Println(timesTwo)
+
 	// TODO: Create an array with 10 different strings in it.
 	// Using this array, create a new array containing the length of each
 	// string.
 	// E.g. ["hello", "!"] => [5, 1]
 	// Hint: Use a loop and the `len` function.
 	// Print the new array.
 
+	var stringArray [10]string
+	stringArray[0] = "So..."
+	stringArray[1] = "how"
+	stringArray[2] = "are"
+	stringArray[3] = "you"
+	stringArray[4] = "holding"
+	stringArray[5] = "up..."
+	stringArray[6] = "BECAUSE"
+	stringArray[7] = "IM"
+	stringArray[8] = "A"
+	stringArray[9] = "POTATO"
+	fmt.Println(stringArray)
+	fmt.Println("len test=", len(stringArray[0]))
+
+	var lengthStringArray [10]int
+	var count3 = 0
+
+	for {
+		if count3 > 9 {
```

Comment:
Use the length of the array here.

Reply:
Yeah that bugged me, had more like this and was on my TODO list. Done.